### DISCUSSION TITLE
Creating a ranger scratchpad in i3wm

### DISCUSSION THREAD

**Discussion by cppt** [i3/i3#5600] [Q&A] [answered]
Have poured over the i3 user manual and assorted tutorials but can't for the life of me get ranger file to launch into a scratchpad.  

Relevant config looks like this (have modified keybindings so there's no conflict here):
```
for_window [instance="ranger"] move scratchpad
exec --no-startup-id alacritty -e ranger -name ranger
bindsym $mod+n [instance="ranger"] scratchpad show
```
From here, have tried both reloading and restarting, as well as logging out/back in without success.  What am I doing wrong?

**Comment by okraits** [comment]
You could use:

* `alacritty --class Alacritty,ranger ...`
* `... [class="Alacritty" instance="ranger"] ...`

**Reply by cppt** [reply]
thanks, for this...would you mind making it explicit?  can't place your first line wrt to my '-e' option specifically

**Reply by okraits** [reply]
```
for_window [class="Alacritty" instance="ranger"] move scratchpad
exec --no-startup-id alacritty -e ranger --class Alacritty,ranger
bindsym $mod+n [class="Alacritty" instance="ranger"] scratchpad show

**Reply by cppt** [reply]
Thanks a bunch...unfortunately, still no dice. 

Also tried lowercase 'a' on Alacritty (as that's what's used to launch), as well as removing 'ranger' after the comma in the line 2 class label.

**Reply by okraits** [reply, marked as answer]
This works here. Just replace `htop` with `ranger`.

`for_window [class="Alacritty" instance="htop"] move scratchpad`
`exec alacritty --class Alacritty,htop -e htop`
`bindsym $mod+n [class="Alacritty" instance="htop"] scratchpad show`

**Reply by cppt** [reply]
That works!  Looks like the exec line was what was screwing things up...thanks a bunch, man!
